### PR TITLE
Checkbox: move from react-next to react-checkbox package

### PR DESCRIPTION
**Pull request by xugao**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes
Prepping `Checkbox` for V8 release, with following steps:
- [x] scaffold new `react-checkbox` and move `Checkbox` from `react-next` to it (current PR)
- [ ] Move current `Checkbox` code to `next` folder and revert default `Checkbox` to only have FC conversion (no scss, use of compose, etc.)

Related issue: #14660

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/react-checkbox/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# @fluentui/react-checkbox
+
+**React Checkbox components for [Fluent UI React](https://developer.microsoft.com/en-us/fluentui)**
+
+These are not production-ready components and **should never be used in product**. This space is useful for testing new components whose APIs might change before final release.
+
+To import React Checkbox components:
+
+```js
+import { Checkbox } from '@fluentui/react-checkbox';
```
Comment (dzearing) [inline packages/react-checkbox/README.md:10]: maybe finish off the example :)

File: packages/react-checkbox/src/components/Checkbox/Checkbox.test.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { create } from '@uifabric/utilities/lib/test';
 import { mount, ReactWrapper } from 'enzyme';
 
 import { Checkbox } from './Checkbox';
-import { IRefObject, resetIds } from '../../Utilities';
+import { IRefObject, resetIds } from '@uifabric/utilities';
```
Comment (khmakoto) [inline packages/react-checkbox/src/components/Checkbox/Checkbox.test.tsx:7]: You can import `@uifabric/styling` and `@uifabric/utilities` from `office-ui-fabric-react/lib/Styling` and `office-ui-fabric-react/lib/Utilities` respectively and remove those dependencies from the `package.json` file.

File: packages/react-checkbox/src/components/Checkbox/examples/Checkbox.Other.Example.tsx [hunk 3]
```diff
@@ -1,7 +1,6 @@
 import * as React from 'react';
-import { Checkbox, ICheckboxProps } from '@fluentui/react-next/lib/Checkbox';
-import { Link } from '@fluentui/react-next/lib/Link';
-import { Stack } from '@fluentui/react-next/lib/Stack';
+import { Link, Stack } from 'office-ui-fabric-react';
```
Comment (khmakoto) [inline packages/react-checkbox/src/components/Checkbox/examples/Checkbox.Other.Example.tsx:2]: Maybe separate into `Link` and `Stack` specifically?

```suggestion
import { Link } from 'office-ui-fabric-react/lib/Link';
import { Stack } from 'office-ui-fabric-react/lib/Stack';
```

File: packages/react-next/src/components/Dropdown/Dropdown.types.ts [hunk 4]
```diff
@@ -6,9 +6,9 @@ import { ISelectableDroppableTextProps } from 'office-ui-fabric-react/lib/utilit
 import { ResponsiveMode } from 'office-ui-fabric-react/lib/utilities/decorators/withResponsiveMode';
 import { IKeytipProps } from '../../Keytip';
 import { RectangleEdge } from 'office-ui-fabric-react/lib/utilities/positioning';
-import { ICheckboxStyleProps, ICheckboxStyles } from '../Checkbox/Checkbox.types';
-import { ILabelStyleProps, ILabelStyles } from 'office-ui-fabric-react/src/components/Label/Label.types';
-import { IPanelStyleProps, IPanelStyles } from 'office-ui-fabric-react/src/components/Panel/Panel.types';
+import { ICheckboxStyleProps, ICheckboxStyles } from 'office-ui-fabric-react/lib/Checkbox';
```
Comment (khmakoto) [inline packages/react-next/src/components/Dropdown/Dropdown.types.ts:9]: Shouldn't this one point to the one in `@fluentui/react-checkbox`?

```suggestion
import { ICheckboxStyleProps, ICheckboxStyles } from '@fluentui/react-checkbox/lib/Checkbox';

```

File: packages/react-checkbox/src/components/Checkbox/Checkbox.test.tsx (hunk 2 above)
Comment (ecraig12345) [reply packages/react-checkbox/src/components/Checkbox/Checkbox.test.tsx:7]: The direct `@uifabric/styling` and `@uifabric/utilities` imports are correct here too for reasons explained in other PR.

**Pull request merged**